### PR TITLE
vmbus channel reassignment and cpu offline test improvement

### PR DESCRIPTION
**Pull request by juhlee-microsoft**
1. Accessed directly to sysfs not using lsvmbus tool.
2. Removed the dmesg/console error verifications
3. Changed the wording

**TEST RESULTS**

[canonical ubuntuserver 18.04-lts Latest] [LISAv2 Test Results Summary]
[canonical ubuntuserver 18.04-lts Latest] Test Run On           : 06/01/2020 00:57:16
[canonical ubuntuserver 18.04-lts Latest] ARM Image Under Test  : canonical : ubuntuserver : 18.04-lts : Latest
[canonical ubuntuserver 18.04-lts Latest] Initial Kernel Version: 5.3.0-1022-azure
[canonical ubuntuserver 18.04-lts Latest] Final Kernel Version  : 5.7.0-rc2+
[canonical ubuntuserver 18.04-lts Latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[canonical ubuntuserver 18.04-lts Latest] Total Time (dd:hh:mm) : 0:0:26
[canonical ubuntuserver 18.04-lts Latest] 
[canonical ubuntuserver 18.04-lts Latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[canonical ubuntuserver 18.04-lts Latest] -------------------------------------------------------------------------------------------------------------------------------------------
[canonical ubuntuserver 18.04-lts Latest]     1 CORE                 CPU-OFFLINE-VMBUS-INTERRUPT-REASSINGMENT                                          PASS                23.73 

[canonical ubuntuserver 16.04-lts Latest] [LISAv2 Test Results Summary]
[canonical ubuntuserver 16.04-lts Latest] Test Run On           : 06/01/2020 00:57:09
[canonical ubuntuserver 16.04-lts Latest] ARM Image Under Test  : canonical : ubuntuserver : 16.04-lts : Latest
[canonical ubuntuserver 16.04-lts Latest] Initial Kernel Version: 4.15.0-1083-azure
[canonical ubuntuserver 16.04-lts Latest] Final Kernel Version  : 5.7.0-rc2+
[canonical ubuntuserver 16.04-lts Latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[canonical ubuntuserver 16.04-lts Latest] Total Time (dd:hh:mm) : 0:0:32
[canonical ubuntuserver 16.04-lts Latest] 
[canonical ubuntuserver 16.04-lts Latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[canonical ubuntuserver 16.04-lts Latest] -------------------------------------------------------------------------------------------------------------------------------------------
[canonical ubuntuserver 16.04-lts Latest]     1 CORE                 CPU-OFFLINE-VMBUS-INTERRUPT-REASSINGMENT                                          PASS                29.69 

[canonical ubuntuserver 18.04-lts Latest] [LISAv2 Test Results Summary]
[canonical ubuntuserver 18.04-lts Latest] Test Run On           : 06/01/2020 01:29:39
[canonical ubuntuserver 18.04-lts Latest] ARM Image Under Test  : canonical : ubuntuserver : 18.04-lts : Latest
[canonical ubuntuserver 18.04-lts Latest] Initial Kernel Version: 5.3.0-1022-azure
[canonical ubuntuserver 18.04-lts Latest] Final Kernel Version  : 5.7.0-rc2+
[canonical ubuntuserver 18.04-lts Latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[canonical ubuntuserver 18.04-lts Latest] Total Time (dd:hh:mm) : 0:1:0
[canonical ubuntuserver 18.04-lts Latest] 
[canonical ubuntuserver 18.04-lts Latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[canonical ubuntuserver 18.04-lts Latest] -------------------------------------------------------------------------------------------------------------------------------------------
[canonical ubuntuserver 18.04-lts Latest]     1 CORE                 STRESS-CPU-OFFLINE-ONLINE                                                         PASS                57.13 

[canonical ubuntuserver 16.04-lts Latest] [LISAv2 Test Results Summary]
[canonical ubuntuserver 16.04-lts Latest] Test Run On           : 06/01/2020 01:29:35
[canonical ubuntuserver 16.04-lts Latest] ARM Image Under Test  : canonical : ubuntuserver : 16.04-lts : Latest
[canonical ubuntuserver 16.04-lts Latest] Initial Kernel Version: 4.15.0-1083-azure
[canonical ubuntuserver 16.04-lts Latest] Final Kernel Version  : 5.7.0-rc2+
[canonical ubuntuserver 16.04-lts Latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[canonical ubuntuserver 16.04-lts Latest] Total Time (dd:hh:mm) : 0:1:38
[canonical ubuntuserver 16.04-lts Latest] 
[canonical ubuntuserver 16.04-lts Latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[canonical ubuntuserver 16.04-lts Latest] -------------------------------------------------------------------------------------------------------------------------------------------
[canonical ubuntuserver 16.04-lts Latest]     1 CORE                 STRESS-CPU-OFFLINE-ONLINE                                                         PASS                96.36

### CODE REVIEW

**Review by aparri — APPROVED** [review]

**Review by aparri — APPROVED** [review]

File: Testscripts/Linux/channel_change.sh [hunk 1]
```diff
@@ -19,229 +13,160 @@
...
-				FailedCount=$((FailedCount+1))
+				# Now reset this cpu id to 0.
+				LogMsg "Set the vmbus channel $_vmbus_ch's cpu to the default cpu, 0"
+				echo 0 > $_device/channels/$_vmbus_ch/cpu
+				sleep 1
+				# validate the change of that cpu id
+				_cpu_id=$(cat $_device/channels/$_vmbus_ch/cpu)
+				if [ $_cpu_id = 0 ]; then
+					LogMsg "Successfully set the vmbus channel $_vmbus_ch's cpu to 0"
+					cpu_idx=$((cpu_idx+1))
```
Comment (aparri) [inline Testscripts/Linux/channel_change.sh]: "cpu_idx" is only used here, IIUC, the line can be removed

File: Testscripts/Linux/channel_change.sh [hunk 2]
```diff
@@ -19,229 +13,160 @@
...
-	LogMsg "VMBUS scanning result: ${vmbus_id[@]}"
+			sleep 1
+		done < channel_vp_mapping
+	done
+	LogMsg "The list of target CPU: ${idle_cpus[@]}"
 
 	# #######################################################################################
+	# The previous step sets all channels' cpu to 0, so the rest of non-zero cpu can be offline
 	# Select a CPU number where does not associate to vmbus channels from idle_cpus array
+	LogMsg "Changed all cpu state from online to offline, and vice versa."
```
Comment (aparri) [inline Testscripts/Linux/channel_change.sh]: Maybe do s/Changed/Change/ (we're going to change the state)

**Review by aparri — APPROVED** [review]

**Review by aparri — DISMISSED** [review]

File: Testscripts/Linux/channel_change.sh (hunk 2 above)
Comment (juhlee-microsoft) [reply Testscripts/Linux/channel_change.sh]: Fixed in 6b1643883a8b1ad8eb082a0e161c107e1bdd9efb

File: Testscripts/Linux/channel_change.sh (hunk 1 above)
Comment (juhlee-microsoft) [reply Testscripts/Linux/channel_change.sh]: Fixed in 6b1643883a8b1ad8eb082a0e161c107e1bdd9efb

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**